source/slang/ir-legalize-types.cpp

```diff
@@ -1244,16 +1305,10 @@ static LegalVal legalizeLocalVar(
 
         context->insertBeforeLocalVar = irLocalVar;
 
-        LegalVarChain* varChain = nullptr;
-        LegalVarChain varChainStorage;
-        if (varLayout)
-        {
-            varChainStorage.next = nullptr;
-            varChainStorage.varLayout = varLayout;
-            varChain = &varChainStorage;
-        }
+        LegalVarChainLink varChain(LegalVarChain(), varLayout);
```

Comment:
On first glance looked worry some, because 'parent' goes out of scope before varChain. I see it takes a copy so ok, but perhaps deserves a comment.